packages/flutter/lib/src/material/data_table.dart

```diff
@@ -635,6 +636,15 @@ class DataTable extends StatelessWidget {
   /// around the table defined by [decoration].
   final bool showBottomBorder;
 
+  /// {@template flutter.material.dataTable.checkboxHorizontalMargin}
+  /// Horizontal margin between the edges of the table and the checkbox as well as
+  /// the content in the first data column, when checkbox is displayed.
```

Comment:
Can you make this change so that the `checkboxHorizontalMargin` is independent of the `horizontalMargin`.

as I understand it here, we will have

```
checkboxHorizontalMargin | Checkbox | checkboxHorizontalMargin / 2 | horizontalMargin / 2 | firstCellContent | horionztalMargin
```

The usage I was thinking of was to make sure that we have it like this:

```
checkboxHorizontalMargin | Checkbox | checkboxHorizontalMargin | horizontalMargin  | firstCellContent | horionztalMargin
```

This means you must update the following lines too:
https://github.com/flutter/flutter/blob/66fabe01c8a12a88e221a4166b337f6ec8848868/packages/flutter/lib/src/material/data_table.dart#L966